--- /frameworks/base/services/java/com/android/server/power/DisplayPowerController.java
+++ /frameworks/base/services/java/com/android/server/power/DisplayPowerController.java
@@ -442,7 +442,7 @@
         if (!DEBUG_PRETEND_PROXIMITY_SENSOR_ABSENT) {
             mProximitySensor = mSensorManager.getDefaultSensor(Sensor.TYPE_PROXIMITY);
             if (mProximitySensor != null) {
-                mProximityThreshold = Math.min(mProximitySensor.getMaximumRange(),
+                mProximityThreshold = Math.min(TYPICAL_PROXIMITY_THRESHOLD,
                         TYPICAL_PROXIMITY_THRESHOLD);
             }
         }
